app/src/auth/components/SessionProvider/index.jsx

```diff
@@ -0,0 +1,63 @@
+// @flow
+
+import React, { type Node } from 'react'
+
+import Context, { type Props as State } from '$auth/contexts/Session'
+import { isSessionStorageAvailable } from '$mp/utils/storage'
+
+const SESSION_TOKEN_KEY = 'session.token'
+
+type Props = {
+    children: Node,
+}
+
+const storage = isSessionStorageAvailable() ? sessionStorage : null
+
+class SessionProvider extends React.Component<Props, State> {
+    static token(): ?string {
+        if (storage) {
+            return storage.getItem(SESSION_TOKEN_KEY) || null
+        }
+        return null
+    }
+
+    static storeToken(value?: ?string) {
+        if (storage) {
+            if (!value) {
+                storage.removeItem(SESSION_TOKEN_KEY)
+            } else {
+                storage.setItem(SESSION_TOKEN_KEY, value)
+            }
+        }
+    }
+
+    state = {
+        setSessionToken: this.setSessionToken.bind(this),
```

Comment:
if you make setSessionToken an arrow function you don't have to worry about binding.

```js
setSessionToken = () => {
  // …
}
```

Reply:
This would require `state` defininition to be placed after `setSessionToken` definition cause order metters. I feel like the current approach causes less confusion and lets us avoid bypassing `react/sort-comp`.